File: test-rest-api/src/com/inform8/entities/LineItem.java
package com.inform8.entities;

public class LineItem {
	private float price = 0;
	private float priceExtGst = 0;
	private int quantity = 0;
	private float subtotal = 0;
	private float subtotalExtGst = 0;
	private ProductOption productOption = null;
	
	public void setSubtotalExtGst(float subtotalExtGst) {
		this.subtotalExtGst = subtotalExtGst;
	}
	public float getSubtotalExtGst() {
		return subtotalExtGst;
	}
	public void setSubtotal(float subtotal) {
		this.subtotal = subtotal;
	}
	public float getSubtotal() {
		return subtotal;
	}
	public void setQuantity(int quantity) {
		this.quantity = quantity;
	}
	public int getQuantity() {
		return quantity;
	}
	public void setPriceExtGst(float priceExtGst) {
		this.priceExtGst = priceExtGst;
	}
	public float getPriceExtGst() {
		return priceExtGst;
	}
	public void setPrice(float price) {
		this.price = price;
	}
	public float getPrice() {
		return price;
	}
	public void setProductOption(ProductOption productOption) {
		this.productOption = productOption;
	}
	public ProductOption getProductOption() {
		return productOption;
	}
	
}

File: rest-api/site/config/lib/RetailDb/shoppingbag/LineItem.php
<?php
/**
 * Represents a Line Item ...
 * @author Ryan Henderson
 * @copyright 2011 - Inform8
 *
 */
 class LineItem {
    
    private $quantity = 0;
    private $productOption;
    
    function __construct($productOption, $quantity) {
      $this->productOption = $productOption;
      $this->quantity = $quantity;
    }

    public function addQuantity($quantity) {
      $this->quantity += $quantity;
    }
    
    public function setQuantity($quantity) {
      $this->quantity = $quantity;
    }
    
    /**
     * @return ProductOption
     */
    public function getProductOption() {
      return $this->productOption;
    }

    public function getQuantity() {
      return $this->quantity;
    }
    
    public function getPrice() {
      $cost = floatval($this->productOption->getWholesalePrice());
      $specialCost = floatval($this->productOption->getSaleWholesalePrice());
      
      if($specialCost != NULL && $specialCost > 0) {
       return $specialCost;
      }
      return $cost;
    }
    
    public function getPriceExGst() {
      $total = $this->getPrice();
      $gst = floatval($total) / floatval(11);
      return round($total-$gst, 2);
    }      

    public function getSubtotal() {
      return $this->getQuantity() * $this->getPrice();
    }
    
    public function getSubtotalExGst() {
      $total = $this->getSubtotal();
      $gst = floatval($total) / floatval(11);
      return round($total-$gst, 2);
    }     
    
    
}